modules/fedimint-dummy-common/src/lib.rs

```diff
@@ -4,23 +4,103 @@ use async_trait::async_trait;
 use fedimint_core::config::ModuleGenParams;
 use fedimint_core::core::{Decoder, ModuleInstanceId, ModuleKind};
 use fedimint_core::encoding::{Decodable, Encodable};
+use fedimint_core::epoch::SerdeSignatureShare;
 use fedimint_core::module::{CommonModuleGen, ModuleCommon, ModuleConsensusVersion};
-use fedimint_core::plugin_types_trait_impl_common;
+use fedimint_core::{plugin_types_trait_impl_common, Amount};
+use secp256k1::XOnlyPublicKey;
 use serde::{Deserialize, Serialize};
 use thiserror::Error;
 
+// Common contains types shared by both the client and server
+
+// The client and server configuration
 pub mod config;
-pub mod db;
 
+/// Unique name for this module
 const KIND: ModuleKind = ModuleKind::from_static_str("dummy");
+
+/// Needed to avoid API path collisions
+// TODO: Remove
+pub const LEGACY_HARDCODED_INSTANCE_ID_DUMMY: ModuleInstanceId = 4;
+
+/// Modules are non-compatible with older versions
 pub const CONSENSUS_VERSION: ModuleConsensusVersion = ModuleConsensusVersion(0);
 
+/// Non-transaction items that will be submitted to consensus
 #[derive(Debug, Clone, Eq, PartialEq, Hash, Serialize, Deserialize, Encodable, Decodable)]
-pub struct DummyConsensusItem;
+pub enum DummyConsensusItem {
+    /// User's print money request signed by a peer
+    Print(DummyPrintMoneyRequest, SerdeSignatureShare),
+}
+
+/// Parameters necessary to generate this module's configuration
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct DummyConfigGenParams {
+    pub tx_fee: Amount,
+}
+
+/// Input for a fedimint transaction
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Deserialize, Serialize, Encodable, Decodable)]
+pub struct DummyInput {
+    pub amount: Amount,
+    /// Associate the input with a user's pubkey
+    pub account: XOnlyPublicKey,
+}
```

Comment:
I think it would be prudent to include some sort of nonce, otherwise a two separate transfers between the same accounts of the same value would be identical (same txid) and thus the second would not go through. This won't happen for a while, but when it does it will be hard to debug.